tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -2485,6 +2543,9 @@ Status StorageManager::load_fragment_metadata(
           metadata->load(encryption_key, f_buff, offset), tdb_delete(metadata));
       open_array->insert_fragment_metadata(metadata);
     }
+    auto array_schema_name = metadata->array_schema_name();
+    auto frag_array_schema = open_array->get_array_schema(array_schema_name);
+    metadata->set_array_schema(frag_array_schema.get());
```

Comment:
We need a null check here to make sure we aren't setting the schema to null because we couldn't find it. We should error in that case.

Reply:
added the null check